Add GitHub issues template and related system report script

frenchwr opened the pull request:
This adds issues template and a system reporting script to help us to better support users of this repo and triage issues.

There may be other system details that would be helpful here. For example, is it possible to get the tdx module version from the CLI (I briefly searched around but did not come across anything)? Or are there other packages that should be included?

For reference, I'm pasting the results of running the system report script from a production system below. Some notes about the output:

* I personally think it's most convenient (both for users and maintainers) to have the report directly in the body of the issue rather than uploading a file. 
* I also considered a bulleted list rather than entire sections but I found the section-based approach easier to read/parse.

 
### Operating system details

```
Distributor ID:	Ubuntu
Description:	Ubuntu 24.04 LTS
Release:	24.04
Codename:	noble
```

### Kernel version

```
6.8.0-1009-intel #16-Ubuntu SMP PREEMPT_DYNAMIC Thu Jul 18 15:46:59 UTC 2024 x86_64 x86_64 GNU/Linux
```

### TDX kernel logs

```
[    0.000000] Command line: BOOT_IMAGE=/boot/vmlinuz-6.8.0-1009-intel root=UUID=2def89e0-5f60-4b15-9fc3-40df79380b94 ro kvm_intel.tdx=1 nohibernate
[    0.651972] Kernel command line: BOOT_IMAGE=/boot/vmlinuz-6.8.0-1009-intel root=UUID=2def89e0-5f60-4b15-9fc3-40df79380b94 ro kvm_intel.tdx=1 nohibernate
[    1.378492] virt/tdx: BIOS enabled: private KeyID range [64, 128)
[    1.378496] virt/tdx: Disable ACPI S3. Turn off TDX in the BIOS to use ACPI S3.
[    8.317823] virt/tdx: TDX module: attributes 0x0, vendor_id 0x8086, major_version 1, minor_version 5, build_date 20240129, build_num 698
[    8.317827] virt/tdx: CMR: [0x100000, 0x77800000)
[    8.317828] virt/tdx: CMR: [0x100000000, 0x206e000000)
[    8.317828] virt/tdx: CMR: [0x2080000000, 0x4070000000)
[    9.015808] virt/tdx: 1050636 KB allocated for PAMT
[    9.015813] virt/tdx: module initialized
```

### TDX CPU instruction support

```
CPU supports TDX according to /proc/cpuinfo
```

### CPU details

```
 Intel(R) Xeon(R) Platinum 8488C
```

### Production system check

```
CPU: Sapphire Rapids
Production
```

### QEMU package details

```
Status: Installed
Package: qemu-system-x86
Version: 1:8.2.2+ds-0ubuntu2+tdx1.0~tdx1.202407191424~ubuntu24.04.1
APT-Sources: https://ppa.launchpadcontent.net/kobuk-team/tdx/ubuntu noble/main amd64 Packages
```

### Libvirt package details

```
Status: Installed
Package: libvirt-clients
Version: 10.0.0-2ubuntu8.3-tdx1.1+tdx.202407260818~ubuntu24.04.1
APT-Sources: https://ppa.launchpadcontent.net/kobuk-team/tdx/ubuntu noble/main amd64 Packages
```

### OVMF package details

```
Status: Installed
Package: ovmf
Version: 2024.02-2+ppa2+tdx.202407031341~ubuntu24.04.1
APT-Sources: https://ppa.launchpadcontent.net/kobuk-team/tdx/ubuntu noble/main amd64 Packages
```

### sgx-dcap-pccs package details

```
Status: Installed
Package: sgx-dcap-pccs
Version: 1.21-1~202407240841~ubuntu24.04.1
APT-Sources: https://ppa.launchpadcontent.net/kobuk-team/tdx/ubuntu noble/main amd64 Packages
```

### tdx-qgs package details

```
Status: Installed
Package: tdx-qgs
Version: 1.21-1~202407220704~ubuntu24.04.1
APT-Sources: https://ppa.launchpadcontent.net/kobuk-team/tdx/ubuntu noble/main amd64 Packages
```

### sgx-ra-service package details

```
Status: Installed
Package: sgx-ra-service
Version: 1.21-1~202407220704~ubuntu24.04.1
APT-Sources: https://ppa.launchpadcontent.net/kobuk-team/tdx/ubuntu noble/main amd64 Packages
Description: Intel(R) Software Guard Extensions Multi-Package Registration Agent Service
```

### sgx-pck-id-retrieval-tool package details

```
Status: Installed
Package: sgx-pck-id-retrieval-tool
Version: 1.21-1~202407220704~ubuntu24.04.1
APT-Sources: https://ppa.launchpadcontent.net/kobuk-team/tdx/ubuntu noble/main amd64 Packages
```

hector-cao commented on .github/ISSUE_TEMPLATE/support.md:
```diff
@@ -0,0 +1,14 @@
+---
+name: Support Request
+about: Support request or question relating to Intel Device Plugins for Kubernetes
```
Is that a typo here ?

frenchwr replied:
Oops - good catch. :) Fixed.

hector-cao reviewed (changes requested):
Great work ! Thanks

hector-cao commented on system-report.sh:
```diff
@@ -0,0 +1,102 @@
+print_section "TDX kernel logs" "${result}"
+
+result=$( \
+if grep -q tdx /proc/cpuinfo; then \
```
see if possible to dump relevant MSR values (see `test_host_tdx_hardware.py`) but it will need sudo ...

frenchwr replied:
I'll check

frenchwr replied:
Other quick note in case it causes confusion, I had a typo in the line above that caused this result to state that the CPU did not contain TDX support. I have since fixed the typo and edited the original PR description to include the updated output.

frenchwr replied:
@hector-cao I added a new section with the MSR values.

hector-cao reviewed (changes requested):
Great work ! Thanks

hector-cao commented:
@frenchwr could you link this PR to the issue #186 ?

frenchwr commented:
This PR addresses https://github.com/canonical/tdx/issues/186

frenchwr commented:
> @frenchwr could you link this PR to the issue #186 ?

done!

bktan8 commented:
@frenchwr @hector-cao - what do you think about adding these additional information?
sudo systemctl status qgsd
sudo systemctl status pccs
cat /var/log/mpa_registration.log

frenchwr commented:
@bktan8 I've updated the report with the information you requested. I'll copy the output below so you can see a sample. I agree with @hector-cao it would be good to provide a description for the new MSRs. For now, I'm just using generic `MSR_EXTRA` labels in the report.

### Operating system details

```
Distributor ID:	Ubuntu
Description:	Ubuntu 24.04 LTS
Release:	24.04
Codename:	noble
```

### Kernel version

```
6.8.0-1009-intel #16-Ubuntu SMP PREEMPT_DYNAMIC Thu Jul 18 15:46:59 UTC 2024 x86_64 x86_64 GNU/Linux
```

### TDX kernel logs

```
[    0.000000] Command line: BOOT_IMAGE=/boot/vmlinuz-6.8.0-1009-intel root=UUID=2def89e0-5f60-4b15-9fc3-40df79380b94 ro kvm_intel.tdx=1 nohibernate
[    0.651972] Kernel command line: BOOT_IMAGE=/boot/vmlinuz-6.8.0-1009-intel root=UUID=2def89e0-5f60-4b15-9fc3-40df79380b94 ro kvm_intel.tdx=1 nohibernate
[    1.378492] virt/tdx: BIOS enabled: private KeyID range [64, 128)
[    1.378496] virt/tdx: Disable ACPI S3. Turn off TDX in the BIOS to use ACPI S3.
[    8.317823] virt/tdx: TDX module: attributes 0x0, vendor_id 0x8086, major_version 1, minor_version 5, build_date 20240129, build_num 698
[    8.317827] virt/tdx: CMR: [0x100000, 0x77800000)
[    8.317828] virt/tdx: CMR: [0x100000000, 0x206e000000)
[    8.317828] virt/tdx: CMR: [0x2080000000, 0x4070000000)
[    9.015808] virt/tdx: 1050636 KB allocated for PAMT
[    9.015813] virt/tdx: module initialized
```

### TDX CPU instruction support

```
CPU supports TDX according to /proc/cpuinfo
```

### Model specific registers (MSRs)

```
HW_ENCRYPT_ENABLE bit: 1 (expected value: 1)
SEAM_RR bit: 1 (expected value: 1)
NUM_TDX_PRIV_KEYS: 40 (expected value: >32)
MSR_EXTRA1 (0xa0): 0 (expected value: 0)
MSR_EXTRA2 (0x1f5, bit 11): 1 (expected value: 1)
MSR_EXTRA3 (0x1401, bit 11): 1 (expected value: 1)
```

### CPU details

```
 Intel(R) Xeon(R) Platinum 8488C
```

### Production system check

```
CPU: Sapphire Rapids
Production
```

### QEMU package details

```
Status: Installed
Package: qemu-system-x86
Version: 1:8.2.2+ds-0ubuntu2+tdx1.0~tdx1.202407191424~ubuntu24.04.1
APT-Sources: https://ppa.launchpadcontent.net/kobuk-team/tdx/ubuntu noble/main amd64 Packages
```

### Libvirt package details

```
Status: Installed
Package: libvirt-clients
Version: 10.0.0-2ubuntu8.3-tdx1.1+tdx.202407260818~ubuntu24.04.1
APT-Sources: https://ppa.launchpadcontent.net/kobuk-team/tdx/ubuntu noble/main amd64 Packages
```

### OVMF package details

```
Status: Installed
Package: ovmf
Version: 2024.02-2+ppa2+tdx.202407031341~ubuntu24.04.1
APT-Sources: https://ppa.launchpadcontent.net/kobuk-team/tdx/ubuntu noble/main amd64 Packages
```

### sgx-dcap-pccs package details

```
Status: Installed
Package: sgx-dcap-pccs
Version: 1.21-1~202407240841~ubuntu24.04.1
APT-Sources: https://ppa.launchpadcontent.net/kobuk-team/tdx/ubuntu noble/main amd64 Packages
```

### tdx-qgs package details

```
Status: Installed
Package: tdx-qgs
Version: 1.21-1~202407220704~ubuntu24.04.1
APT-Sources: https://ppa.launchpadcontent.net/kobuk-team/tdx/ubuntu noble/main amd64 Packages
```

### sgx-ra-service package details

```
Status: Installed
Package: sgx-ra-service
Version: 1.21-1~202407220704~ubuntu24.04.1
APT-Sources: https://ppa.launchpadcontent.net/kobuk-team/tdx/ubuntu noble/main amd64 Packages
Description: Intel(R) Software Guard Extensions Multi-Package Registration Agent Service
```

### sgx-pck-id-retrieval-tool package details

```
Status: Installed
Package: sgx-pck-id-retrieval-tool
Version: 1.21-1~202407220704~ubuntu24.04.1
APT-Sources: https://ppa.launchpadcontent.net/kobuk-team/tdx/ubuntu noble/main amd64 Packages
```

### QGSD service status

```
Warning: The unit file, source configuration file or drop-ins of qgsd.service changed on disk. Run 'systemctl daemon-reload' to reload units.
● qgsd.service - Intel(R) TD Quoting Generation Service
     Loaded: loaded (/usr/lib/systemd/system/qgsd.service; enabled; preset: enabled)
     Active: active (running) since Fri 2024-08-02 06:32:43 UTC; 1 week 0 days ago
   Main PID: 61552 (qgs)
      Tasks: 5 (limit: 308220)
     Memory: 632.0K (peak: 3.5M)
        CPU: 80ms
     CGroup: /system.slice/qgsd.service
             └─61552 /usr/bin/qgs

Aug 02 06:32:43 curtis-739457 systemd[1]: Starting qgsd.service - Intel(R) TD Quoting Generation Service...
Aug 02 06:32:43 curtis-739457 systemd[1]: Started qgsd.service - Intel(R) TD Quoting Generation Service.
Aug 02 06:32:43 curtis-739457 qgsd[61552]: Added signal handler
Aug 02 06:32:43 curtis-739457 qgsd[61552]: About to create QgsServer with num_thread = 4
Aug 02 06:32:43 curtis-739457 qgsd[61552]: About to start main loop
```

### PCCS service status

```
Warning: The unit file, source configuration file or drop-ins of pccs.service changed on disk. Run 'systemctl daemon-reload' to reload units.
○ pccs.service - Provisioning Certificate Caching Service (PCCS)
     Loaded: loaded (/usr/lib/systemd/system/pccs.service; disabled; preset: enabled)
     Active: inactive (dead)
       Docs: https://github.com/intel/SGXDataCenterAttestationPrimitives/blob/master/QuoteGeneration/pccs/README.md
```

### MPA registration logs (last 30 lines)

```
[23-07-2024 07:13:22] INFO: Registration Flow - Registration status indicates registration is completed successfully. MPA has nothing to do.
[23-07-2024 07:13:22] INFO: Finished Registration Agent Flow.
[23-07-2024 07:50:41] INFO: SGX Registration Agent version: 1.21.100.3
[23-07-2024 07:50:41] INFO: Starts Registration Agent Flow.
[23-07-2024 07:50:41] INFO: Registration Flow - Registration status indicates registration is completed successfully. MPA has nothing to do.
[23-07-2024 07:50:41] INFO: Finished Registration Agent Flow.
[23-07-2024 05:56:29] INFO: SGX Registration Agent version: 1.21.100.3
[23-07-2024 05:56:29] INFO: Starts Registration Agent Flow.
[23-07-2024 05:56:30] INFO: Registration Flow - PLATFORM_ESTABLISHMENT or TCB_RECOVERY passed successfully.
[23-07-2024 05:56:30] INFO: Finished Registration Agent Flow.
[24-07-2024 02:37:35] INFO: SGX Registration Agent version: 1.21.100.3
[24-07-2024 02:37:35] INFO: Starts Registration Agent Flow.
[24-07-2024 02:37:35] INFO: Registration Flow - Registration status indicates registration is completed successfully. MPA has nothing to do.
[24-07-2024 02:37:35] INFO: Finished Registration Agent Flow.
[30-07-2024 06:48:30] INFO: SGX Registration Agent version: 1.20.100.2
[30-07-2024 06:48:30] INFO: Starts Registration Agent Flow.
[30-07-2024 06:48:30] INFO: Registration Flow - Registration status indicates registration is completed successfully. MPA has nothing to do.
[30-07-2024 06:48:30] INFO: Finished Registration Agent Flow.
[30-07-2024 01:08:07] INFO: SGX Registration Agent version: 1.20.100.2
[30-07-2024 01:08:07] INFO: Starts Registration Agent Flow.
[30-07-2024 01:08:07] INFO: Registration Flow - Registration status indicates registration is completed successfully. MPA has nothing to do.
[30-07-2024 01:08:07] INFO: Finished Registration Agent Flow.
[01-08-2024 06:26:41] INFO: SGX Registration Agent version: 1.20.100.2
[01-08-2024 06:26:41] INFO: Starts Registration Agent Flow.
[01-08-2024 06:26:41] INFO: Registration Flow - Registration status indicates registration is completed successfully. MPA has nothing to do.
[01-08-2024 06:26:41] INFO: Finished Registration Agent Flow.
[02-08-2024 06:32:19] INFO: SGX Registration Agent version: 1.21.100.3
[02-08-2024 06:32:19] INFO: Starts Registration Agent Flow.
[02-08-2024 06:32:19] INFO: Registration Flow - Registration status indicates registration is completed successfully. MPA has nothing to do.
[02-08-2024 06:32:19] INFO: Finished Registration Agent Flow.
```

bktan8 commented:
LGTM

bktan8 commented:
> @bktan8 good idea, for the msr values, can you provide the semantic of different expected values ?

Check the Intel SGX and MCHECK status.  The expected output is 0.
$ sudo rdmsr 0xa0

Check the Intel TDX Status.  The expected output is 1.
$ sudo rdmsr -f 11:11 0x1401

Let's not use 0x1f5 afterall.

frenchwr commented:
@bktan8 @hector-cao Updated. The output for the section on curtis now looks like this:

### Model specific registers (MSRs)

```
HW_ENCRYPT_ENABLE bit: 1 (expected value: 1)
SEAM_RR bit: 1 (expected value: 1)
NUM_TDX_PRIV_KEYS: 40 (expected value: >32)
SGX_AND_MCHECK_STATUS: 0 (expected value: 0)
TDX_STATUS bit: 1 (expected value: 1)
```

hector-cao commented:
@bktan8 @frenchwr SEAM_RR and TDX_STATUS is the same thing, we already read this status and we call it SEAM_RR after the official Intel documentation

frenchwr commented:
@hector-cao Great catch. I removed the duplicate MSR. Here's the new output:

### Model specific registers (MSRs)

```
HW_ENCRYPT_ENABLE bit: 1 (expected value: 1)
SEAM_RR bit: 1 (expected value: 1)
NUM_TDX_PRIV_KEYS: 40 (expected value: >32)
SGX_AND_MCHECK_STATUS: 0 (expected value: 0)
```

hector-cao reviewed (approved):
LGTM; great work; thanks
